src/librustc/ty/maps/plumbing.rs

```diff
@@ -329,13 +351,6 @@ macro_rules! define_maps {
                     };
                     mem::drop(lock);
 
-                    // This just matches the behavior of `try_get_with` so the span when
-                    // we await matches the span we would use when executing.
-                    // See the FIXME there.
-                    if span == DUMMY_SP && stringify!($name) != "def_span" {
-                        span = key.default_span(tcx);
-                    }
-
```

Comment:
Ooh, so much nicer! `:)`

Reply:
I forgot the remove the other copy of this code in `try_get_with`, that will be gone too.

Reply:
Remove the `[WIP]` from the title once you're ready for merging.